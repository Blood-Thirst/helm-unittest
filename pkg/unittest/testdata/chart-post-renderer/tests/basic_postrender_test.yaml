templates:
  - templates/basic.yaml
postRenderer:
  cmd: "yq"
  args:
    - "eval"
    - '.metadata.annotations.appended="new"'
    - "-"
tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: {}
  - it: manifest should include an extra annotation due to suite yq post-renderer
    asserts:
      - equal:
          path: metadata.annotations.foo
          value: bar
      - equal:
          path: metadata.annotations.appended
          value: new
  - it: manifest should overwrite existing annotation due to test-job yq post-renderer
    postRenderer:
      cmd: "yq"
      args:
        - "eval"
        - '.metadata.annotations.foo="baz"'
        - "-"
    asserts:
      - equal:
          path: metadata.annotations.foo
          value: baz
      # never added by the suite yq because it was overridden in the job
      - notExists:
          path: metadata.annotations.appended
