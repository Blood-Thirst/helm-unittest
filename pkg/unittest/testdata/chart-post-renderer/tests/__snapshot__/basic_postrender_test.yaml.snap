manifest should match snapshot:
  1: |
    metadata:
      annotations:
        appended: new
        foo: bar
